src/libs/use-analytics.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+
+const { init, sendPageview } = require('./ga-analytics');
+
+export default function useGoogleAnalytics() {
+  const location = useLocation();
+
+  React.useEffect(() => {
+    init();
+  }, []);
+
+  React.useEffect(() => {
+    const currentPath = location.pathname + location.search;
+    sendPageview(currentPath);
+  }, [location]);
```

Comment:
Oh, nice! I thought `location` is mutable but I was wrong. Tested in the sandbox: https://codesandbox.io/s/relaxed-cloud-9l0lq?file=/src/App.js 👍